src/main/java/in/koreatech/koin/domain/timetable/service/TimetableService.java

```diff
@@ -40,58 +49,90 @@ public List<LectureResponse> getLecturesBySemester(String semester) {
-        Integer grades = timeTables.stream()
-            .mapToInt(timeTable -> Integer.parseInt(timeTable.getGrades()))
+        List<Integer> lectureIds = timetableLecture.stream().map(tl -> tl.getLecture().getId()).toList();
+        List<Lecture> lectures = lectureRepository.findAllByIds(lectureIds);
+
+        Integer grades = lectures.stream()
+            .mapToInt(lecture -> Integer.parseInt(lecture.getGrades()))
             .sum();
         Integer totalGrades = calculateTotalGrades(userId);
 
-        return TimeTableResponse.of(semester.getSemester(), timeTables, grades, totalGrades);
+        return TimetableResponse.of(semester.getSemester(), timetableLecture, lectures, grades, totalGrades);
     }
 
     private int calculateTotalGrades(Integer userId) {
         int totalGrades = 0;
         List<Semester> semesters = semesterRepository.findAllByOrderBySemesterDesc();
 
         for (Semester semester : semesters) {
-            totalGrades += timeTableRepository.findAllByUserIdAndSemesterId(userId, semester.getId()).stream()
+            totalGrades += timetableRepository.findAllByUserIdAndSemesterId(userId, semester.getId()).stream()
                 .mapToInt(timeTable -> Integer.parseInt(timeTable.getGrades()))
                 .sum();
         }
 
         return totalGrades;
     }
+
+    @Transactional
+    public void deleteTimetableLecture(Integer id) {
+        timetableLectureRepository.getById(id);
+        timetableLectureRepository.deleteById(id);
+    }
+
+    @Transactional
+    public TimetableFrameUpdateResponse updateTimetableFrame(Integer timetableFrameId,
+        TimetableFrameUpdateRequest timetableFrameUpdateRequest, Integer userId) {
+        TimetableFrame timeTableFrame = timetableFrameRepository.getById(timetableFrameId);
+        Semester semester = semesterRepository.getBySemester(timetableFrameUpdateRequest.semester());
+        boolean isMain = timetableFrameUpdateRequest.isMain();
+        if (isMain) {
+            cancelMainTimetable(userId, semester.getId());
+        }
+        timeTableFrame.updateTimetableFrame(semester, timetableFrameUpdateRequest.name(),
+            timetableFrameUpdateRequest.isMain());
+        return TimetableFrameUpdateResponse.from(timeTableFrame);
+    }
+
+    private void cancelMainTimetable(Integer userId, Integer semesterId) {
+        TimetableFrame mainTimetableFrame = findMainTimetable(userId, semesterId);
+        mainTimetableFrame.cancelMain();
+    }
+
+    private TimetableFrame findMainTimetable(Integer userId, Integer semesterId) {
+        return timetableFrameRepository.findAllByUserIdAndSemesterId(userId, semesterId)
+            .stream()
+            .filter(TimetableFrame::isMain)
+            .findFirst()
+            .orElseThrow(() -> TimetableLectureNotFoundException.withDetail("못 찾음 userId: "+ userId + ", semesterId: " + semesterId));
+    }
```

Comment:
```suggestion
private void cancelMainTimetable(Integer userId, Integer semesterId) {
    TimetableFrame mainTimetableFrame = timetableFrameRepository.findAllByUserIdAndSemesterId(userId, semesterId)
        .stream()
        .filter(TimetableFrame::isMain)
        .findFirst()
        .orElseThrow(() -> new TimetableLectureNotFoundException("못 찾음 userId: " + userId + ", semesterId: " + semesterId));
    mainTimetableFrame.cancelMain();
}
```

A
얘네가 재 사용 될 로직이 아니라면 하나로 합치는 것이 더 좋을 수도 있을 것 같습니당

Reply:
getTimetableResponse()에서도 findMainTimetable()를 사용하고 있습니다!
 \+ 하나의 메서드에 역할 한 개만 가지고 있는 게 좋다 생각해서 나눠봤습니다